.github/workflows/release.yml

```diff
@@ -0,0 +1,61 @@
+name: Build and Push
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  commitlint:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+        with:
+          fetch-depth: 0
+      - name: Validate HEAD Commit
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: |
+          npm i
+          npx commitlint --from HEAD~1 --to HEAD --verbose
+
+  build:
+    needs: commitlint
+    runs-on: ubuntu-22.04
+    if: ${{ contains(fromJSON('["!:", "BREAKING CHANGE:", "patch", "minor", "major"]'), github.event.head_commit.message) || !(startsWith('chore', github.event.head_commit.message) || startsWith('refactor', github.event.head_commit.message) || startsWith('docs', github.event.head_commit.message)) }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+
+      - name: Login to GHCR
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Tag Version
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: |
+          npm i
```

Comment:
Any reason to go with npm, because we are favoring pnpm now.

Reply:
It is an extra thing to install, and this is such a simple repo - no workspaces with complex deps.